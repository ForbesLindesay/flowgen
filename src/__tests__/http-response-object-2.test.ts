import {transformFile} from '../';

test('index.d.ts', () => {
  expect(transformFile(__dirname + '/http-response-object-2/index.d.ts')).toMatchSnapshot();
});
test('headers.d.ts', () => {
  expect(transformFile(__dirname + '/http-response-object-2/headers.d.ts')).toMatchSnapshot();
});