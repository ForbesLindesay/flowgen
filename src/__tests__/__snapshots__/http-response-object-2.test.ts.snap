// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`headers.d.ts 1`] = `
"// @flow
// Generated using flowgen2

export type Headers = {+[name: string]: string | Array<string>};
"
`;

exports[`index.d.ts 1`] = `
"// @flow
// Generated using flowgen2

import type {Headers} from './headers';

declare class Response<TBody> {
  +statusCode: number,
  +headers: Headers,
  +body: TBody,
  +url: string,
  constructor(
    statusCode: number,
    headers: Headers,
    body: TBody,
    url: string,
  ): void,
  getBody(encoding: string): string,
  getBody(): TBody,
}

module.exports = Response;
"
`;
